awx/main/tasks.py

```diff
@@ -765,7 +765,7 @@ def build_extra_vars_file(self, vars, **kwargs):
         handle, path = tempfile.mkstemp(dir=kwargs.get('private_data_dir', None))
         f = os.fdopen(handle, 'w')
         if settings.ALLOW_JINJA_IN_EXTRA_VARS == 'always':
-            f.write(yaml.safe_dump(vars))
+            f.write(safe_dump(vars, vars))
```

Comment:
Do we need this?  Seems like a lot of unnecessary comparison considering they match.

Reply:
_thinking very hard_

`safe_dump(vars, None)` would seem to mark everything as unsafe, and in this case, we want to mark nothing as unsafe. That's because `None` is interpreted as meaning that there's no safe vars reference structure.

https://github.com/ansible/awx/blob/f4ac8e8645bbadcfa3d0f468a217d5bcc881035c/awx/main/utils/safe_yaml.py#L115

If we _just_ did `yaml.safe_dump(vars)`, then that loses the vault representer, which is the source of the linked issue for this PR.

However, it was lazy / hacky to use the existing pattern from the other cases here. Want we really want for this particular case might be more like

```python
f.write(yaml.dump_all([vars], dumper=UnsafeDumper, default_flow_style=False))
```

Where `UnsafeDumper` is a dumper that has the vault representer, but does not have the unsafe string representer.

Looking at the related utils now, I realize I may have already implemented this wrong. If the particular key-value par is safe, then the custom AWX `safe_dump` method will use `yaml.SafeDumper`, which I _think_ never had the vault representer added, so presumably this dumper will probably encounter a `VaultData` object that the constructor produced, and then error.

Side note: I can't manage to pass the dumper via `yaml.safe_dump`, but you might have already hit this frustration in your original implementation of `safe_dump`.